### ISSUE TITLE
log_grad_norm should use sync_dist=True

### ISSUE THREAD

**Issue by lendle** [Lightning-AI/lightning#16084]
### Bug description

`LightningModule.log_grad_norm` calls `self.log_dict` without setting `sync_dist=True`. This produces a bunch of warnings when training with a distributed strategy. 
Example:
```
PossibleUserWarning: It is recommended to use `self.log('grad_2.0_norm/<PARAMETER_NAME>', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
```

### How to reproduce the bug

```python
trainer = Trainer(..., track_grad_norm = 2, strategy = "ddp_notebook")
trainer.fit(model)
```

### Error messages and logs

```
PossibleUserWarning: It is recommended to use `self.log('grad_2.0_norm/<PARAMETER_NAME>', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
```
repeated for every parameter at the end of each epoch

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0): 1.8.1. I know this isn't the newest release but it's still present in the master branch
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 1.10): 1.13
#- Python version (e.g., 3.9): 3.8
#- OS (e.g., Linux): Debian
#- CUDA/cuDNN version: 11.6
#- GPU models and configuration: 
#- How you installed Lightning(`conda`, `pip`, source): conda
#- Running environment of LightningApp (e.g. local, cloud): 
```

</details>

### More info

I'm happy to provide a full example to reproduce if necessary, but I believe (perhaps naively) that this is pretty simple to follow!

As a workaround I'm overriding `log_grad_norm` in my `LightningModule` with
```
    def log_grad_norm(self, grad_norm_dict: Dict[str, float]) -> None:

        self.log_dict(grad_norm_dict, on_step=True, on_epoch=True, prog_bar=False, logger=True, sync_dist=True)

```

**Comment by alexpersin** [comment]
I can confirm I can reproduce this issue too.

**Comment by lendle** [comment]
> This issue has been automatically marked as stale because it hasn't had any recent activity. This issue will be closed in 7 days if no further activity occurs. Thank you for your contributions - the Lightning Team!

This seems to still be an issue. Any way to remove the `won't fix` label?

**Comment by carmocca** [comment]
> As a workaround I'm overriding log_grad_norm in my LightningModule with

The suggested "workaround" in the top post is the intended way to do this in 1.9.x:

Additionally, https://github.com/Lightning-AI/lightning/pull/16745 removed support for `track_grad_norm` (to land in 2.0).
In the PR's description, you can find how to compute the grad_norm and log it manually with your desired logging arguments.

Closing this, as either of the options will let you set the `sync_dist` value to your preference. Thanks!
